client/src/components/LabelInput/LabelInput.tsx

```diff
@@ -1,210 +1,194 @@
 import React from "react";
```

Comment:
`LabelInput` now owns the validation + submission flow for both categories and labels. The autosave wiring relies on `onSubmit` to pass the trimmed value so the thunk decides whether to persist immediately or wait for selection.